### PR TITLE
Issue #233 validator worker docker image

### LINKED ISSUE #233: Docker image
Docker Image:
- ~~`alpine` linux~~ (not possible atm)
- The `validator_worker` binary

Should:
- Be compatible with the JS impl & environment variables

`alpine` is not possible at the moment since:
```
Caused by:
  cannot produce proc-macro for `const-random-macro v0.1.6` as the target `x86_64-unknown-linux-musl` does not support these crate types
```

### PR DESCRIPTION
**Pull request by elpiel**
Fixes #233

### CODE REVIEW

File: adapter/Cargo.toml [hunk 1]
```diff
@@ -19,8 +19,8 @@ eth_checksum = "0.1.1"
 ethabi = { git = "https://github.com/samparsky/ethabi", branch = "graph-patches" }
 tiny-keccak = "1.5"
 parity-crypto = { version = "0.4.2", features = ["publickey"] }
-ethstore = { version = "0.2.1", git = "https://github.com/paritytech/parity-ethereum"}
-ethkey = { version = "0.4.0", git = "https://github.com/paritytech/parity-ethereum"}
+ethstore = { version = "0.2.1", git = "https://github.com/elpiel/parity-ethereum"}
+ethkey = { version = "0.4.0", git = "https://github.com/elpiel/parity-ethereum"}
```
Comment (samparsky) [inline adapter/Cargo.toml]: why the change here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
+FROM rust:1.40 as builder
+
```
Comment (samparsky) [inline Dockerfile]: Nice work! using multi-stage builds

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+COPY . .
+
+RUN cargo install --path validator_worker --all-features
+
+WORKDIR /usr/local/bin
+
+RUN cp $CARGO_HOME/bin/validator_worker .
+
+FROM ubuntu:18.04
```
[the remark this answers is not in the document]
Comment (Ivshti) [reply Dockerfile]: yeah, alpine uses musl so it can't run the rust binaries
@elpiel correct me if I'm wrong

File: adapter/Cargo.toml (hunk 1 above)
Comment (Ivshti) [reply adapter/Cargo.toml]: does it sync faster?

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (elpiel) [reply Dockerfile]: Yes, that's the problem (`openssl` crate doesn't build on musc)

File: adapter/Cargo.toml (hunk 1 above)
Comment (elpiel) [reply adapter/Cargo.toml]: faster - yes
fast build - not so much it still takes long time, but it's better.
I tried looking into caching dependencies, but not much luck on that.
@samparsky sent me a link, so I will check it if it's possible to use cache for the **build**

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (elpiel) [reply Dockerfile]: Sorry, I lied... it's not `openssl`, it was another crate:
```
Caused by:
  cannot produce proc-macro for `const-random-macro v0.1.6` as the target `x86_64-unknown-linux-musl` does not support these crate types
```
